staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -132,14 +141,19 @@ func WithMaxInFlightLimit(
 	var mutatingChan chan bool
 	if nonMutatingLimit != 0 {
 		nonMutatingChan = make(chan bool, nonMutatingLimit)
-		watermark.readOnlyObserver.SetDenominator(float64(nonMutatingLimit))
+		klog.V(2).InfoS("Initialized nonMutatingChan", "len", nonMutatingLimit)
+	} else {
+		klog.V(2).InfoS("Running with nil nonMutatingChan")
 	}
 	if mutatingLimit != 0 {
 		mutatingChan = make(chan bool, mutatingLimit)
-		watermark.mutatingObserver.SetDenominator(float64(mutatingLimit))
+		klog.V(2).InfoS("Initialized mutatingChan", "len", mutatingLimit)
+	} else {
+		klog.V(2).InfoS("Running with nil mutatingChan")
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		initMaxinflight(nonMutatingLimit, mutatingLimit)
```

Comment:
I know that synchronization on `once` is relatively cheap, but still I think we should avoid that in the handler function.
Why not calling it directly from WithMaxInFlightLimit - all metrics should be initialized at this point anyway.

Reply:
I was unsure of that.
I was not confident that this piece of code should rely on that knowledge.

Reply:
done